src/script/descriptor.cpp

```diff
@@ -2039,11 +2059,18 @@ std::vector<std::unique_ptr<DescriptorImpl>> ParseScript(uint32_t& key_exp_index
         for (size_t i = 0; i < max_providers_len; ++i) {
             // Build final subscripts vectors by retrieving the i'th subscript for each vector in subscripts
             std::vector<std::unique_ptr<DescriptorImpl>> this_subs;
+            std::vector<TRNodeInfo> this_nodes;
             this_subs.reserve(subscripts.size());
-            for (auto& subs : subscripts) {
-                this_subs.emplace_back(std::move(subs.at(i)));
+            this_nodes.reserve(subscripts.size());
+            for (size_t pos = 0; pos < subscripts.size(); pos++) {
+                this_subs.emplace_back(std::move(subscripts[pos].at(i)));
+                TRNodeInfo node;
+                node.depth = depths[pos];
+                node.leaf_version = TAPROOT_LEAF_TAPSCRIPT;
+                node.type = TRNodeType::LEAF_SCRIPT;
+                this_nodes.emplace_back(node);
```

Comment:
In  "descriptor: Add TRNodeInfo" 475981527f052f7c9709d1f3a90270770f4dc160

You should add a constructor to `TRNodeInfo`, this will prevent constructing the object and individually adding all elements before emplacing to the vector here and other places.

```suggestion
                this_nodes.emplace_back(depths[pos], TAPROOT_LEAF_TAPSCRIPT, TRNodeType::LEAF_SCRIPT);
```